onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -996,26 +1030,27 @@ static bool PropagateFP32CastsFromInputsToOutputs(Graph& graph, Node* node,
       }
     }
     if (has_float_inputs && (level >= 2 || all_float_inputs_have_casts) && casts.size() > 1) {
-      if (non_cast_producers_map.size() > 0) {
+      if (!non_cast_producers_map.empty()) {
         ORT_THROW_IF_ERROR(InsertCastNodes(graph, non_cast_producers_map, true, removed_nodes, inserted_nodes));
         LOGS(logger, VERBOSE) << "PropagateFP32CastsFromInputsToOutputs: Inserted FP16 Cast node to "
-                              << ConcatNames<NodeArgToConsumerMap>(non_cast_producers_map, GetName);
+                              << ConcatNames(non_cast_producers_map, GetName);
       }
       for (Node* cast : casts) {
         ORT_THROW_IF_ERROR(RemoveCastNodesChain(graph, {cast}, removed_nodes));
       }
       LOGS(logger, VERBOSE) << "PropagateFP32CastsFromInputsToOutputs: Removed Cast nodes "
-                            << ConcatNames<std::vector<Node*>>(casts)
+                            << ConcatNames(casts)
                             << " feeding the same compute node " << node->Name();
       NodeArgToConsumerMap node_args_map;
       for (NodeArg* output : node->MutableOutputDefs()) {
         if (output->Exists() && IsRelevantOutput(node, output) && IsType(*output, TensorProto::FLOAT)) {
-          node_args_map.insert(std::make_pair(output, graph.GetMutableConsumerNodes(output->Name())));
+          auto mutable_consumer_nodes = graph.GetMutableConsumerNodes(output->Name());
```

Comment:
>GetMutableConsumerNodes [](http://example.com/codeflow?start=46&length=23)

I guess it's beyond the scope of this PR, but will we change this one to return an GetMutableConsumerNodes() to return an InlinedVector too?

Reply:
Yes, graph and Session will go later